registry/config.go

```diff
@@ -20,15 +20,15 @@ type ServiceOptions struct {
 	Mirrors                        []string `json:"registry-mirrors,omitempty"`
 	InsecureRegistries             []string `json:"insecure-registries,omitempty"`
 
-	// V2Only controls access to legacy registries.  If it is set to true via the
-	// command line flag the daemon will not attempt to contact v1 legacy registries
+	// V2Only controls access to legacy registries.
```

Comment:
There is a check now that is enforcing this is not false, what is defaulting it to true?

Reply:
The flag is still there, and has `true` as default; https://github.com/moby/moby/pull/35479/files#diff-fe42cd3c297bb829e5e2d79a7640a52dR94

Although not strictly necessary, I thought keeping the flag around (but producing a proper error) was better than removing the flag (and getting an `"unknown flag:  --disable-legacy-registry"`) - same for existing `daemon.json` files